samcli/commands/build/build_context.py

```diff
@@ -263,12 +265,7 @@ def run(self):
                 )
                 output_template_path = stack.get_output_template_path(self.build_dir)
 
-                if self._create_auto_dependency_layer:
-                    LOG.debug("Auto creating dependency layer for each function resource into a nested stack")
-                    nested_stack_manager = NestedStackManager(
-                        stack, self._stack_name, self.build_dir, modified_template, build_result
-                    )
-                    modified_template = nested_stack_manager.generate_auto_dependency_layer_stack()
+                modified_template = self._handle_build_post_processing(stack, modified_template, build_result)
```

Comment:
discussed in huddle: some of the lines above could also go into the post_processing function